### PR TITLE
root: relax libxml2 requirements when ~python.

### PR DESCRIPTION
**Pull request by chissg**
`depends_on('libxml2~python', when='~python')` -> `depends_on('libxml2', when='~python')` per https://github.com/spack/spack/issues/13637#issuecomment-552026561.

### CODE REVIEW

File: var/spack/repos/builtin/packages/root/package.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ class Root(CMakePackage):
     depends_on('veccore',   when='+veccore')
     depends_on('vdt',       when='+vdt')
     depends_on('libxml2+python',   when='+xml+python')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/root/package.py]: I think you want to remove that line, too
```suggestion
```

Comment (chissg) [reply var/spack/repos/builtin/packages/root/package.py]: I *thought* I'd checked that ROOT's Python bindings required libxml2's Python bindings, but my neurons are failing, so I will double-check.

Comment (chissg) [reply var/spack/repos/builtin/packages/root/package.py]: I have confirmed with ROOT devs that ROOT's Python bindings do not in turn require libxml2's Python bindings. I will amend the PR.

Comment (ax3l) [reply var/spack/repos/builtin/packages/root/package.py]: Thanks a lot for checking with upstream!

File: var/spack/repos/builtin/packages/root/package.py [hunk 2]
```diff
@@ -507,23 +506,27 @@ def cmake_args(self):
...
-        spack_env.set('ROOT_VERSION', 'v{0}'.format(self.version.up_to(1)))
-        spack_env.prepend_path('PYTHONPATH', self.prefix.lib)
-        spack_env.prepend_path('PATH', self.prefix.bin)
-        spack_env.append_path('CMAKE_MODULE_PATH', '{0}/cmake'
-                              .format(self.prefix))
-        run_env.set('ROOTSYS', self.prefix)
-        run_env.set('ROOT_VERSION', 'v{0}'.format(self.version.up_to(1)))
-        run_env.prepend_path('PYTHONPATH', self.prefix.lib)
-        run_env.prepend_path('PATH', self.prefix.bin)
+            env.append_path('CMAKE_PREFIX_PATH',
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/root/package.py]: interesting, this should be automatically be set by Spack, no? cc @alalazo

Comment (chissg) [reply var/spack/repos/builtin/packages/root/package.py]: There's an issue if `lz4` is found from `packages.yaml`, I think. It was definitely found to be necessary, and similar things have been done in other recipes.

**Pull request merged**